projects/plugins/jetpack/modules/subscriptions/subscribe-modal/class-jetpack-subscribe-modal.php

```diff
@@ -197,27 +197,11 @@ public function should_user_see_modal() {
 
 		// Don't show if user is subscribed to blog.
 		require_once __DIR__ . '/../views.php';
-		if ( $this->has_subscription_cookie() || Jetpack_Subscriptions_Widget::is_current_user_subscribed() ) {
+		if ( Jetpack_Memberships::is_current_user_subscribed() ) {
```

Comment:
Do we need to ensure the class is loaded, similar to Subscription block paywall?

https://github.com/Automattic/jetpack/blob/8c3dabed53c440f1061cb2fcabc66daf10e384ca/projects/plugins/jetpack/extensions/blocks/subscriptions/subscriptions.php#L792